.github/workflows/ci.yml

```diff
@@ -0,0 +1,73 @@
+name: Rust CI
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  build:
+    # This job tests on Ubuntu and macOS.
+    strategy:
+      matrix:
+        os: [ubuntu-22.04, macos-13]
+        rust:
+          - stable
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: Swatinem/rust-cache@v2
+
+    - name: Install Rust Toolchain (stable)
+      run: |
+          rustup toolchain install stable --profile minimal
```

Comment:
well i think here the rustup toolchain should switch to both the MSRV and 1.68.0, happy to take this change in a follow-up PR

Reply:
I did briefly test this but I agree, lets work on this in a seperate PR. I tried to implement a quicker solution but since this is quite well developed, its best to finish this properly then address that rust version setup. Would you want me to open a new issue?

Reply:
Yes feel free to open a new issue or open pr as you wish.